src/test/java/io/jenkins/security/CompatibilityTest.java

```diff
@@ -4,39 +4,40 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
-import java.net.URL;
+import java.net.URISyntaxException;
 import java.util.Arrays;
 import java.util.Objects;
 import java.util.stream.Stream;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipFile;
-import java.util.zip.ZipInputStream;
 import org.apache.commons.io.IOUtils;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assertions.fail;
 
 public class CompatibilityTest {
 
     private static final String REMOTING_JAR_NAME_PATTERN = "remoting[-](.+)[.]jar";
-    private static final String REMOTING_JAR_DIRECTORY = "target/classes/old-remoting/";
 
-    private static Stream<Arguments> test() {
-        return Arrays.stream(Objects.requireNonNull(new File(REMOTING_JAR_DIRECTORY).list((dir, name) -> name.matches(REMOTING_JAR_NAME_PATTERN)))).map(Arguments::of);
+    private static Stream<Arguments> test() throws URISyntaxException {
+        return Arrays.stream(Objects.requireNonNull(getRemotingDirectory().list((dir, name) -> name.matches(REMOTING_JAR_NAME_PATTERN)))).map(Arguments::of);
     }
     @ParameterizedTest
     @MethodSource
-    public void test(String remotingFileName) throws IOException {
-        final File remotingFile = new File(REMOTING_JAR_DIRECTORY + remotingFileName);
+    public void test(String remotingFileName) throws IOException, URISyntaxException {
+        final File remotingFile = new File(getRemotingDirectory(), remotingFileName);
         assertTrue(remotingFile.exists(), "Failed to find remoting " + remotingFileName);
         final byte[] classfile = readClassfileFromRemoting(new ZipFile(remotingFile));
         assertNotNull(Security3430Workaround.innerTransform(classfile), "Failed to transform class file in remoting " + remotingFileName);
     }
 
+    private static File getRemotingDirectory() throws URISyntaxException {
+        return new File(Objects.requireNonNull(CompatibilityTest.class.getResource("/old-remoting/remoting-4.0.jar")).toURI().getPath()).getParentFile();
```

Comment:
Will probably work on Linux systems with no spaces in the dir name, but in general do not treat `URI.path` as a `File.path`; use `File.<init>(URI)` instead.